### PR TITLE
modal

### LINKED ISSUE #25: Modal

### PR DESCRIPTION
**Pull request by theodorusclarence**
# Description & Technical Solution

Modal, with Modal.Section to customize items

Update website branding, rencananya pengen gw duplicate ke repo gw, nanti mo gw taro di twitter, lumayan portfolio kecil"an

Use manual self host font method because of bugs

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] Already rebased against main branch.

# Screenshots

https://user-images.githubusercontent.com/55318172/212482866-ccdd8046-2d5f-45c0-81ed-90ab95e4cbbf.mp4

- Closes #25

### CODE REVIEW

**Comment by ppdbultimate** [comment]
#### 📝 Changed routes:


| [**/_app.page**](https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/_app.page) | [**/index.page**](https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/index.page) |
|:-:|:-:|
|<a href="https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/_app.page">Screenshot of /_app.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/ZnCXVRO0pd1fk8ZanmCufn1EFpJHnalUge427Emab0vOdYBruQT4Jpc2-HPdGQWRwp3ixz-9eW6ha_uqX_Ndtg.png">(view full size)</a> |<a href="https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/index.page">Screenshot of /index.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/JhRItTobwDZGLFHWJ14QxpDLCtrb3N4hhc0oPIE66ChMVzeNW_bFDdthrL6Nq2ZS6P2rdYoxD2PG3oTlFNgRfw.png">(view full size)</a> |



| [**…omponents/ExampleModal**](https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/sandbox/components/ExampleModal) | [**/sandbox/modal.page**](https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/sandbox/modal.page) |
|:-:|:-:|
|<a href="https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/sandbox/components/ExampleModal">Screenshot of /sandbox/components/ExampleModal</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/s8e6r6sqJYJZxaONeVH4gM5V3gsOWOITZV955qag125tG2WIYgUSG4UlG-CzjjBPRh-PMCTJcmxIgA1CXyUtFw.png">(view full size)</a> |<a href="https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app/sandbox/modal.page">Screenshot of /sandbox/modal.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/bt7fLkl-7tpY-b75mH9nd4ETfMLP9ubZaHbYutmUrjM5S3BnenY_iO0DbI3C0hg4Cr6Dv4OCbLa0zZgLIo2XIQ.png">(view full size)</a> |







Commit f1007e68576a0a05074053ab392bee22f6cf4f51 (https://design-system-moon-bxz6xk7s4-ppdbultimate.vercel.app).

**Comment by theodorusclarence** [comment]
[fix: next font is not exposed to whole app](https://github.com/ppdbultimate/design-system-moon/pull/26/commits/6911acf6ad4ff843894edd4a1e823d0d6e41d7da)

kayaknya approach ini ada problem, karena tadi abis nyoba, dia pas initial visit ke page, malah jadi ga keload fontnya.
harus dicek lagi

File: src/components/modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+/**
+ * defaults to py-6, change it with className if needed
+ */
+function Section({
+  className,
+  children,
+}: React.ComponentPropsWithoutRef<'div'>) {
+  return (
+    <div
+      className={clsxm('flex w-full flex-col px-4 sm:px-6', 'py-6', className)}
```
Comment (winatungmiharja) [inline src/components/modal/Modal.tsx:120]: perlu ditambahkan default `p-4 sm:p-6` mas? kalau defaultnya di mobile soalnya kurang simetris yang bagian section nya
image

sedangkan bagian title nya baguss
image

File: src/components/modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+function Section({
+  className,
+  children,
+}: React.ComponentPropsWithoutRef<'div'>) {
+  return (
+    <div
+      className={clsxm(
+        'flex w-full flex-col px-4 sm:px-6',
+        'py-4 sm:py-6',
+        className
```
Comment (winatungmiharja) [inline src/components/modal/Modal.tsx:123]: ini kalau diubah jadi
`p-4 sm:p-6` gimana mas

Comment (theodorusclarence) [reply src/components/modal/Modal.tsx:123]: okee

File: src/pages/sandbox/components/ExampleModal.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <Modal.Section>
+          <Typography variant='b2'>
+            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Distinctio
+            ex cumque assumenda! Nesciunt reiciendis dolores sit atque incidunt
+            debitis repudiandae.
+          </Typography>
+        </Modal.Section>
+        <Modal.Section>
+          <div className='flex justify-end gap-2'>
+            <Button className='justify-center' variant='outline'>
```
Comment (rizqitsani) [inline src/pages/sandbox/components/ExampleModal.tsx:34]: ini gaperlu justify-center ce, di Button udah ada

File: src/pages/sandbox/components/ExampleModal.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+              Label
+            </Button>
+            <Button className='justify-center'>Label</Button>
```
Comment (rizqitsani) [inline src/pages/sandbox/components/ExampleModal.tsx:37]: ini juga

**Review by rizqitsani — CHANGES_REQUESTED** [review]

**Review by winatungmiharja — APPROVED** [review]

**Review by rizqitsani — APPROVED** [review]

**Pull request merged**